pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2199,6 +2199,14 @@ public void startSendOperation(Producer producer, int msgSize, int numMessages)
             }
             isPublishRateExceeded = producer.getTopic().isBrokerPublishRateExceeded();
         } else {
+            if (producer.getTopic().isResourceGroupRateLimitingEnabled()) {
```

Comment:
Perhaps this makes more sense being in a else if clause?

Reply:
ok.